### ISSUE TITLE
Truesilver breastplate heal does not proc

### ISSUE THREAD

**Issue by SuqMadiqM8** [slowtorta/turtlewow-bug-tracker#4136]
**Please, provide a clear description what the bug is:**
The truesilver breastplate has a 3% chance to heal for 60-100 hp when struck in combat. Yet after about ten hours of using it for aoe grinding on a paladin (usually fighting 3-4 mobs at a time) it appears I have not received a single heal from the item. Combat log and floating combat text (no addons used) show all other procs of 'when struck' effects and show all heals of other sources, which leads me to the conclusion that truesilver breastplate is bugged.

**Steps to reproduce the behavior with as much detail as possible:**

1. Equip truesilver breastplate
2. Get hit by mobs a lot
3. Receive no heals from it

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**

Item: https://database.turtle-wow.org/?item=7939

**Expected behavior. Describe how it should work:**
Having the item equipped you should receive a heal roughly every 33rd +- time you're struck in combat. Yes that's not how % works technically.

**Comment by Dalzedur** [comment]
For what it is worth, I too have been using the said breastplate and leveled to 60 with it. It did proc for me but very rarely, I am not actually so sure that those procs ever reached 3% proc chance, so maybe some other effect is interfering with that chance being checked in the first place. As in, a check has to pass - or fail - and in one of these outcomes the code goes and checks for the next 'when struck' effect in the stack, but not in the other case. So if this is the case, it would mean that the realized proc chance could be way lower than 3%. My truesilver bplate had a lesser absorbtion enchantment (5% chance on when struck to get 25 pts of absorb shield and that enchantment procced in almost every encounter, the heal in maybe one in twenty)

Anyways, 3% proc chance is, in layman terms, 3 hits out of 100 or 1 hit out of 33 (you can easily get hit over 33 times in a battle engagement if you are a low dps but high survivability pally, and that's like a bell curve chance of it happening, as it might happen multiple times or none at all, too). And by hit, specifically hits that don't miss, don't get parried, dodged - not sure if shield block either partially or by fully negating the hit prevents the check from going through. This might be the number one culprit to look at, since prot paladins tend to have a lot of shield block chance with redoubt and holy shield having decent uptime when aoe farming. The first point of 'confusion' is to actually figure out if the hit has to deal at least 1 point of hitpoint damage in order for the effect to be rolled for.

So basically, you might have shadow objects stacked over the player object with all having the function with the same name (query_when_struck(), for example), and the one on top - last one to shadow the shadowed player object - (checked first) returns 1 or 0 or something else instead of 'return shadowed_object()->query_when_struck() which would make the check go through to the next object and its query_when_struck() function when an effect happens or does not happen. This combobreaks the multiple query_when_struck() effects that should normally/logically get all checked, in order, one at a time. The query_when_struck() functions below this first return of 'something else but what is needed for the roll to continue' don't get checked properly.

But I've never looked under the bonnet of the private server WoW, so I'm basically just relying on my past experience with an LPC multi-user dungeon and how it would handle a case like this.

**Comment by SuqMadiqM8** [comment]
Thanks for you insight on the matter. I currently have the Girdle of reprisal (https://database.turtle-wow.org/?item=11861) and Uther's Strength (https://database.turtle-wow.org/?item=11302) equipped and will test if either of them might interfere with the 'when struck' of the truesilver breastplate.

**Comment by SuqMadiqM8** [comment]
Have tested with and without other 'when struck' effect items equipped. With Uther's Strength equipped the breastplate does not proc. Without it equipped it does occasionally proc and combat log shows 'Your Self Healing healed you for xx.'

When you equip the breastplate and the trinket it does not proc, when you unequip the trinket it will still not proc. I also had to take of the breastplate and equip it again to get it to work. Maybe I'll still get the heals if I change the order in which I equip trinket and breastplate (last).

**Comment by Dalzedur** [comment]
Yeah, order of wearing them can matter depending on the code involved. If you have time and energy, you could also test if not using a shield has a significant effect on how often you get the procs, in case holy shield and redoubt are also lowering the chance.

**Comment by balakethelock** [comment]
Why was this closed as completed? @SuqMadiqM8 
If Truesilver Breastplate's proc is turned off by equipping Uther's Strength then it is indeed bugged.

**Comment by D4nSch** [comment]
> Why was this closed as completed? @SuqMadiqM8 If Truesilver Breastplate's proc is turned off by equipping Uther's Strength then it is indeed bugged.

Uther's Strength grants you a shield on procc (https://www.wowhead.com/classic/spell=8397/uthers-light), while this is active hits don't count as strucks and the Truesilver Breastplate (https://www.wowhead.com/classic/spell=9778/proc-self-heal-22) therefore doesn't procc!

Imagine it like warriors not gaining rage on hits while being affected by Powerword:Shield of a priest. Same behavior!

**Comment by Dalzedur** [comment]
Sigh, I'll try and be polite. The shield given by Uther's Strength does not have 100% uptime. This means there are periods of time - even long periods of time - when the shield just isn't active. Why then would the truesilver breastplate not proc even when the shield from Uther's Strength is down.
